integrations/jira/jira/client.py

```diff
@@ -103,107 +180,95 @@ async def create_events_webhook(self, app_host: str) -> None:
+        )
 
-    async def get_paginated_users(
-        self,
-    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+    async def get_paginated_users(self) -> AsyncGenerator[list[dict[str, Any]], None]:
         logger.info("Getting users from Jira")
+        async for users in self._get_paginated_data(f"{self.api_url}/users/search"):
+            yield users
 
-        params = self._generate_base_req_params()
-
-        total_users = len(await self._get_users_data(params))
-
-        if total_users == 0:
-            logger.warning(
-                "User query returned 0 users, did you provide the correct Jira API credentials?"
-            )
-
-        params["maxResults"] = PAGE_SIZE
-        while params["startAt"] < total_users:
-            logger.info(f"Current query position: {params['startAt']}/{total_users}")
-
-            user_response_list = await self._get_users_data(params)
-
-            if not user_response_list:
-                logger.warning(f"No users found at {params['startAt']}")
-                break
+    async def get_paginated_teams(
+        self, org_id: str
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        logger.info("Getting teams from Jira")
 
-            logger.info(
-                f"Retrieved users: {len(user_response_list)} "
-                f"(Position: {params['startAt']}/{total_users})"
-            )
+        base_url = f"{self.teams_base_url}/{org_id}/teams"
 
-            yield user_response_list
-            params["startAt"] += PAGE_SIZE
+        async for teams in self._get_cursor_paginated_data(
+            url=base_url, method="GET", extract_key="entities", cursor_param="cursor"
+        ):
+            yield teams
 
-    async def get_single_user(self, account_id: str) -> dict[str, Any]:
-        user_response = await self.client.get(
-            f"{self.api_url}/user", params={"accountId": account_id}
-        )
-        user_response.raise_for_status()
-        return user_response.json()
+    async def get_paginated_team_members(
+        self, team_id: str, org_id: str, page_size: int = PAGE_SIZE
+    ) -> AsyncGenerator[list[dict[str, Any]], None]:
+        url = f"{self.teams_base_url}/{org_id}/teams/{team_id}/members"
+
+        async for members in self._get_cursor_paginated_data(
+            url,
+            method="POST",
+            extract_key="results",
+            initial_params={"first": page_size},
+            cursor_param="after",
+        ):
+            yield members
+
+    async def enrich_teams_with_members(
+        self, teams: list[dict[str, Any]], org_id: str
+    ) -> list[dict[str, Any]]:
+        logger.debug(f"Fetching members for {len(teams)} teams")
+
+        async def fetch_team_members(team_id: str) -> list[dict[str, Any]]:
+            members = []
+            async for batch in self.get_paginated_team_members(team_id, org_id):
+                members.extend(batch)
```

Comment:
I see what you mean - we're consuming from an async generator (get_paginated_team_members) so yielding would maintain that pattern. However, we need to collect the results into a list here because we're using asyncio.gather() for parallel fetching of team members.